### PR TITLE
Feat/home screen UI

### LINKED ISSUE #18: Home Screen UI
The Home Screen Should Contain:
- The trips: ideally in chronological order
- A button that will be used to add a trip

### PR DESCRIPTION
**Pull request by 1andreea26**
Closes #18
This pull request finalises the Overview Screen which now correctly fetches trips from the Firestore Repository and displays their name. Furthermore, I added default values to the Trip attributes and Location attributes (to be able to use the toObject to convert the document). Trips are also sorted by their start date.
This is both a new feature and a fix.

- Screens/UI:  Updates Overview Screen to display upcoming trips.
- Improvements: Default values to Trip attributes.

- Tests: The Overview Screen has been tested to display all correct components using UI tests and manual testing.

Edit:
- Screens/UI: Now on button press user is redirected to AddTrip screen.
- Fix: The init method of the repository now does not call the collection.get() because it was causing a bug which was not loading the Trips for the user after logging in.

### CODE REVIEW

**Comment by g0g05arui** [comment]
@1andreea26 Looks good, quick question though: what are the `.DS_store` binary files you introduced?

**Comment by g0g05arui** [comment]
and what is the purpose of adding default values for the trip data class?

**Comment by g0g05arui** [comment]
Also the build seems to be failing, please check the `lateinit` variables in the tests

**Comment by 1andreea26** [comment]
> @1andreea26 Looks good, quick question though: what are the `.DS_store` binary files you introduced?
Sorry, they're Mac system files. I added them to gitignore to avoid pushing them next time.

**Comment by 1andreea26** [comment]
> and what is the purpose of adding default values for the trip data class?

I added them to have default values in case the user doesn't input something. Checking again, for the moment it's not possible but in the future maybe the user wants to leave for example the description field empty.

**Comment by 1andreea26** [comment]
> Also the build seems to be failing, please check the `lateinit` variables in the tests

Fixed.

File: app/src/androidTest/java/com/android/voyageur/ui/overview/OverviewScreenTest.kt [hunk 1]
```diff
@@ -3,24 +3,33 @@ package com.android.voyageur.ui.overview
...
 
   @Before
   fun setUp() {
+    tripRepository = mock(TripRepository::class.java)
     navigationActions = mock(NavigationActions::class.java)
+    tripViewModel = TripsViewModel(tripRepository)
     `when`(navigationActions.currentRoute()).thenReturn(Route.OVERVIEW)
-    composeTestRule.setContent { OverviewScreen(navigationActions) }
+    composeTestRule.setContent { OverviewScreen(tripViewModel, navigationActions) }
   }
```
Comment (ioanajianu) [inline app/src/androidTest/java/com/android/voyageur/ui/overview/OverviewScreenTest.kt:34]: Could you also add displayTextWhenNoTrips test? There's something similar done in the bootcamp, it tests that the empty prompt is displayed when no trips were added.

Comment (1andreea26) [reply app/src/androidTest/java/com/android/voyageur/ui/overview/OverviewScreenTest.kt:36]: Thanks for the suggestion! I added the test, and added another one to verify it also displays card items.

Comment (ioanajianu) [reply app/src/androidTest/java/com/android/voyageur/ui/overview/OverviewScreenTest.kt:36]: The tests look good, thanks for adding them!

**Review by g0g05arui — APPROVED** [review]
Looks good, all the comments were addressed. I will aprove this now.
